extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts

```diff
@@ -214,7 +214,7 @@ export class VariantAnalysisManager
     }
   }
 
-  private async runVariantAnalysisFromPublishedPack(): Promise<void> {
+  public async runVariantAnalysisFromPublishedPack(): Promise<void> {
```

Comment:
Yes, this would be better. I'd forgotten and thought it was more effort than this to set up the real command manager, but this is fine.

Reply:
Ok, turns out this doesn't work perfectly cleanly because then it uses the `VariantAnalysisManager` that's different from the instance we create in the test setup, so our mocking doesn't work.

For now I'm just going to revert this change back to how it was and have the method be public. Long term I'm not sure what to do here. Not mocking `runVariantAnalysis` has issues too because it makes it much harder to tell that it's doing the filtering of non-problem queries. You get a lot less insight into what's going on internally and only really tell that "a" variant analysis was triggered. At least that's the way it looks to me right now. Maybe it can work ok without too many issues.

Reply:
Ah yeah good point. Okay I have a few ideas around this but lets come back to it later on!